synapse/app/homeserver.py

```diff
@@ -677,6 +677,39 @@ def profiled(*args, **kargs):
         ThreadPool._worker = profile(ThreadPool._worker)
         reactor.run = profile(reactor.run)
 
+    start_time = hs.get_clock().time()
+
+    @defer.inlineCallbacks
+    def phone_stats_home():
+        now = int(hs.get_clock().time())
+        uptime = int(now - start_time)
+        if uptime < 0:
+            uptime = 0
+
+        stats = {}
+        stats["homeserver"] = hs.config.server_name
+        stats["timestamp"] = now
+        stats["uptime_seconds"] = uptime
+        stats["total_users"] = yield hs.get_datastore().count_all_users()
+
+        all_rooms = yield hs.get_datastore().get_rooms(False)
+        stats["total_room_count"] = len(all_rooms)
+
+        stats["daily_active_users"] = yield hs.get_datastore().count_daily_users()
+        daily_messages = yield hs.get_datastore().count_daily_messages()
+        if daily_messages is not None:
+            stats["daily_messages"] = daily_messages
+
+        logger.info("Reporting stats to matrix.org: %s" % (stats,))
+        hs.get_simple_http_client().put_json(
+            "https://matrix.org/report-usage-stats/push",
+            stats
+        )
```

Comment:
Something helpful:

2015-09-22 13:11:23,634 - twisted - 154 - CRITICAL - - Unhandled error in Deferred:


Traceback (most recent call last):
  File "/home/wadaniel/.virtualenvs/synapsemacaroons/local/lib/python2.7/site-packages/twisted/internet/defer.py", line 501, in _startRunCallbacks
    self._runCallbacks()
  File "/home/wadaniel/.virtualenvs/synapsemacaroons/local/lib/python2.7/site-packages/twisted/internet/defer.py", line 588, in _runCallbacks
    current.result = callback(current.result, *args, **kw)
  File "synapse/util/logcontext.py", line 169, in g
    res = f(res, *args, **kwargs)
  File "/home/wadaniel/.virtualenvs/synapsemacaroons/local/lib/python2.7/site-packages/twisted/internet/defer.py", line 1184, in gotResult
    _inlineCallbacks(r, g, deferred)
--- <exception caught here> ---
  File "/home/wadaniel/.virtualenvs/synapsemacaroons/local/lib/python2.7/site-packages/twisted/internet/defer.py", line 1128, in _inlineCallbacks
    result = g.send(result)
  File "synapse/http/client.py", line 223, in put_json
    raise CodeMessageException(response.code, body)
synapse.api.errors.CodeMessageException: 400: {"error_message": "unable to process request"}


Something *clean*, not necessarily. Would you like an explicit errback or something?